### PR TITLE
chore(ci): use python 3.9

### PR DESCRIPTION
**Pull request by chenrui333**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

The formulae is buildable with [python 3.9](https://github.com/Homebrew/homebrew-core/pull/67440), hence update the CI to pointing to the latest version.

### CODE REVIEW

**Comment by schosterbarak** [comment]
i think that enforcing python 3.9 will increase the barrier of entrance for checkov users that are using python 3.8 or 3.7. 
would they still be able to install checkov for that version?
It's also defined that way on the setup.py https://github.com/bridgecrewio/checkov/blob/master/setup.py#L56

**Comment by metahertz** [comment]
Linking this issue as @robeden had some concerns w/3.9 #688

**Comment by chenrui333** [comment]
I don't see any problem of using the latest python (based on my usage as well as homebrew formulae), in fact the hybrid python versions would generally cause more confusions and some issues in the runtime.

If the end use do want to switch between different python versions, they can accomplish that by using something like `pyenv` and setup independent working environment.

Let me know if that makes sense.

**Comment by robeden** [comment]
_Supporting_ 3.9 is great, but I wouldn't be able to run checkov at the moment if it _required_ it. Using it in the GH actions seems fine, but I'd rather not go there for the Pip requirements.

File: Pipfile [hunk 1]
```diff
@@ -37,4 +37,4 @@ semantic_version = "*"
 packaging = "*"
 
 [requires]
-python_version = "3.7"
+python_version = "3.9"
```
Comment (robeden) [inline Pipfile:40]: Recommend: ">=3.7" or ">=3.8"

Comment (schosterbarak) [reply Pipfile:40]: @chenrui333  i agree with Rob here. can you please modify that?

**Comment by schosterbarak** [comment]
Closing due to inactivity

File: Pipfile (hunk 1 above)
Comment (chenrui333) [reply Pipfile:40]: I see, somehow this got lost in my box.

Comment (chenrui333) [reply Pipfile:40]: @schosterbarak can you reopen the PR and I can add the change?

Comment (schosterbarak) [reply Pipfile:40]: @chenrui333 sure. you might need to rebase due conflicts

**Comment by chenrui333** [comment]
doing the change now.

File: Pipfile (hunk 1 above)
Comment (chenrui333) [reply Pipfile]: absolutely.

**Comment by antonblr** [comment]
If you don't want stick with particular python version, you could use job matrix for tests, something like:

```yaml
jobs:
  integration-tests:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: ['3.7', '3.8', '3.9']     
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: ${{ matrix.python-version }}
   ...
```

**Comment by nimrodkor** [comment]
@chenrui333 Can we see this PR as resolved, as today checkov supports 3.7, 3.8 and 3.9?

**Comment by njgibbon** [comment]
Hello,

I think we should be able to have a satisying closure on this given the work done and state of things as summarised below:

1. Given the matrix testing implemented this year Checkov can confidently say it supports use with Python 3.7, 3.8 and 3.9.

2. The setup.py file indicates this which is what the pip tool uses when downloading checkov from PyPi.
```
python_requires=">=3.7"
```
(Note there is no "<=3.9" because python 3.10 is not publicly released yet. And therefor this statement is semantically correct as checkov supports all versions after 3.7 After python 3.10 is released it  would make sense to test compatibility and such and perhaps change the setup.py version in some way)

3. The pipfile syntax is set as follows: 
```
[requires]
python_version = "3.7"
```
This is used if creating a pyenv environment.
The pipfile does not support greater than or less than suntax. You can set the python_version to '3' to indicate all versions of 3 are supported but this would not be accurate, as we do only ensure we support 3.7, 3.8 and 3.9. So, the above is correct. It makes sense to use the lowest version you support for development because there is less chance you will use some non-backwards compatible dependency. As we do matrix testing, an argument could still be made for using the latest you support (3.9) because anything backwards compatible would be caught by that. Anyway, the core point here is what currently there is makes sense and >= syntax is not supported here.

I encourage anyone interested in this to read this thread and specifically this comment to get a good grasp of this: https://github.com/pypa/pipenv/issues/1050#issuecomment-361485919

In looking into this I found one little thing - https://github.com/bridgecrewio/checkov/pull/1331

**Pull request closed without merging**